packages/web-components/src/rating-display/rating-display.ts

```diff
@@ -0,0 +1,90 @@
+import { attr, FASTElement, nullableNumberConverter } from '@microsoft/fast-element';
+import { uniqueId } from '@microsoft/fast-web-utilities';
+import { RatingDisplayColor, RatingDisplaySize } from './rating-display.options.js';
+
+/**
+ * The base class used for constructing a fluent-rating-display custom element
+ * @public
+ */
+export class RatingDisplay extends FASTElement {
+  /**
+   * The color of the rating display icons.
+   *
+   * @public
+   * @default `marigold`
+   * @remarks
+   * HTML Attribute: `color`
+   */
+  @attr
+  public color: RatingDisplayColor = RatingDisplayColor.marigold;
+
+  /**
+   * Renders a single filled icon with a label next to it.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `compact`
+   */
+  @attr({ attribute: 'compact', mode: 'boolean' })
+  public compact: boolean = false;
+  protected compactChanged(): void {
+    if (this.compact) {
+      this.max = 1;
+    }
+  }
+
+  /**
+   * The number of ratings.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `count`
+   */
+  @attr({ converter: nullableNumberConverter })
+  public count?: number;
+
+  /**
+   * The maximum possible value of the rating.
+   * This attribute determines the number of icons displayed.
+   * Must be a whole number greater than 1.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `max`
+   */
+  @attr({ converter: nullableNumberConverter })
+  public max: number = 5;
+  protected maxChanged(): void {
+    if (this.compact) {
+      this.max = 1;
```

Comment:
This might be my ignorance about FAST, but does this potentially fire multiple times? If I set `compact=true` and that sets `max = 1`, does `maxChanged` get called? And if `maxChanged` sets `max = 1`... does `maxChanged` get called again?

Reply:
Given this is just for the calculations in the template, I'd prefer we not produce side effects and change attributes which are passed - I think that will be confusing. Instead of this, can we use a getter and guard that way?

Reply:
This is not needed anymore since I've moved the calculations to the class. We have access to the attributes needed for the calculations from inside the class.